code/modules/mob/living/carbon/carbon.dm

```diff
@@ -650,16 +650,23 @@
 		if(health<= HEALTH_THRESHOLD_DEAD || !getorgan(/obj/item/organ/brain))
 			death()
 			return
-		if(paralysis || sleeping || getOxyLoss() > 50 || (status_flags & FAKEDEATH) || health <= HEALTH_THRESHOLD_CRIT)
+		if(paralysis || sleeping || getOxyLoss() > 50 || (status_flags & FAKEDEATH) || health <= HEALTH_THRESHOLD_DEEPCRIT)
 			if(stat == CONSCIOUS)
 				stat = UNCONSCIOUS
 				blind_eyes(1)
 				update_canmove()
+		else if(health <= HEALTH_THRESHOLD_CRIT)
+			status_flags |= NEARCRIT
+			Weaken(3)
+			update_canmove()

```

Comment:
You need to set the player to `lying = TRUE` as well and modify `/proc/update_canmove()` to simply return `TRUE` if the player is `lying` and has the `NEARCRIT` stat

However, i would suggest moving the `NEARCRIT` stat altogether and just checking the players `health` because we don't need to wait for a `life()` tick to update their canmove